### PR TITLE
RED-113638 - update GO version to 1.21.5

### PR DESCRIPTION
**Pull request by almoggue**

### CODE REVIEW

File: tools.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build tools
 // +build tools
```
Comment (zcahana) [inline tools.go]: I believe the old build constraint can now be removed:
https://stackoverflow.com/questions/68360688/whats-the-difference-between-gobuild-and-build-directives

File: internal/plugin/new_user.go [hunk 2]
```diff
@@ -14,16 +14,18 @@ import (
...
 // If a database configuration exists, the role must be bound to an ACL in the database.
 //
 // or
-// {
-//    "acl" : "acl_name"
-// }
+//   {
+//        "acl" : "acl_name"
+//	  }
+
```
Comment (zcahana) [inline internal/plugin/new_user.go:28]: Accidental?

**Review by zcahana — COMMENTED** [review]
See minor comments inline.
Did you manage to build and test the plugin?

**Comment by almoggue** [comment]
> See minor comments inline. Did you manage to build and test the plugin?

yes, I ran the `make test`
and also used the plugin in our test_vault and it passed

File: internal/plugin/new_user.go (hunk 2 above)
Comment (almoggue) [reply internal/plugin/new_user.go:28]: no, there were formatting errors when I tried to build it without changing the indents

**Review by zcahana — APPROVED** [review]
LGTM

**Pull request merged**